server/routes/users.js

```diff
@@ -93,598 +93,596 @@ router.get('/login', redirectProfile, (req, res) => {
-					userObject.save();
-
-					req.session.destroy(() => {
-						res.clearCookie(process.env.SESS_NAME);
-					});
-
-					res.send({ success: true });
-				} else {
-					return handleError(res, 400, 'Password token expired');
-				}
-			} else {
-				return handleError(res, 400, 'User not found');
-			}
-		} catch (err) {
-			return handleError(res, 400, err);
-		}
-	},
+  '/password/reset/:token',
+  limiter,
+  postPasswordResetValidationRules(),
+  validate,
+  async (req, res) => {
+    try {
+      const userObject = await UserRepository.getUserByPasswordResetToken(req.params.token);
+
+      if (userObject) {
+        if (moment(userObject.passwordResetTokenExpires) > moment()) {
+          const newPassword = await hashPassword(req.body.password);
+          userObject.password = newPassword;
+          userObject.passwordResetToken = null;
+          userObject.passwordResetTokenExpires = null;
+          userObject.save();
+
+          req.session.destroy(() => {
+            res.clearCookie(process.env.SESS_NAME);
+          });
+
+          res.send({ success: true });
+        } else {
+          return handleError(res, 400, 'Password token expired');
+        }
+      } else {
+        return handleError(res, 400, 'User not found');
+      }
+    } catch (err) {
+      return handleError(res, 400, err);
+    }
+  },
 );
 
 // @desc    Get agency details.
 // @access  Private, only users
 router.get('/agency/address', async (req, res) => {
-	try {
-		if (!req.session.user) {
-			return handleError(res, 403, 'No user id in request');
-		}
-		const agencyDetail = await AgencyRepository.getAgencyByUserId(req.session.user._id);
-		res.status(200).send(
-			JSON.stringify({
-				success: true,
-				error: null,
-				data: agencyDetail,
-			}),
-		);
-	} catch (err) {
-		return handleError(res, 400, err);
-	}
-});
+  try {
+    if (!req.session.user) {
+      return handleError(res, 403, 'No user id in request');
+    }
+    const agencyDetail = await AgencyRepository.getAgencyByUserId(req.session.user._id);
+    res.status(200).send(
+      JSON.stringify({
+        success: true,
+        error: null,
+        data: agencyDetail,
+      }),
+    );
+  } catch (err) {
+    return handleError(res, 400, err);
+  }
+},
+);
 
-router.get('/profile/donations', redirectLogin, async (req, res) => {
-	try {
-		const { user } = req.session;
-		let donations;
-		if (user.userRole === 'partner') {
-			const { agency } = req.session;
-			donations = await DonationRepository.getDonationsByAgency(agency._id);
-		} else {
-			donations = await DonationRepository.getDonationsByUser(user._id);
-		}
-		res.render('donationHistory', { donations }, (error, html) => {
-			if (error) {
-				res.status(400).json({ success: false, error });
-			} else {
-				res.status(200).send(html);
-			}
-		});
-	} catch (error) {
-		handleError(res, 400, error);
-	}
+router.get("/profile/donations", redirectLogin, async (req, res) => {
```

Comment:
wrong "?

Reply:
old server, that's from merging the master